src/tokenized-line.coffee

```diff
@@ -59,7 +74,7 @@ class TokenizedLine
     screenColumn = 0
     currentBufferColumn = 0
     for token in @tokens
-      break if currentBufferColumn > bufferColumn
+      break if currentBufferColumn + token.bufferDelta > bufferColumn
```

Comment:
@adzenith I agree to put some boundaries around soft tabs behavior, therefore please go ahead with this :+1:  

By the way, I tested the code below on **master** and, unless we're exactly at the edge of an atomic token, it always clips to left, whereas the new code will clip to right if we're in the middle (thus making these tests fail).

```coffee
      tabLength = 4
      displayBuffer.setTabLength(tabLength)

      expect(displayBuffer.screenPositionForBufferPosition([2, 0])).toEqual [2, 0]
      expect(displayBuffer.screenPositionForBufferPosition([2, 1])).toEqual [2, 0]
      expect(displayBuffer.screenPositionForBufferPosition([2, 2])).toEqual [2, 0]
      expect(displayBuffer.screenPositionForBufferPosition([2, 3])).toEqual [2, 4]
      expect(displayBuffer.screenPositionForBufferPosition([2, 4])).toEqual [2, 4]
```

However, I believe that **master** behavior is not correct and that the change we're making here is the right way to go.

Great job, @adzenith :fire: